classes/form/autocomplete_form.php

```diff
@@ -17,62 +17,80 @@
 /**
  * Implement autocomplete moodle form.
  *
- * @package          tool_wsformat
- * @copyright        2023 Djarran Cotleanu
- * @author           Djarran Cotleanu
- * @license          http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ * @package   tool_wsformat
+ * @copyright 2023 Djarran Cotleanu, Zach Pregl
+ * @author    Djarran Cotleanu, Zach Pregl
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
  */
 
 namespace tool_wsformat\form;
 
+
 use moodleform;
 
 /**
  * Form for selecting web services to format.
  *
  * @package   tool_wsformat
- * @copyright 2023 Djarran Cotleanu
- * @author    Djarran Cotleanu
+ * @copyright 2023 Djarran Cotleanu, Zach Pregl
+ * @author    Djarran Cotleanu, Zach Pregl
  * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
  */
 class autocomplete_form extends moodleform {
 
+
     /**
      * Define an autocomplete element for browsing webservices and a submit button.
      */
-    public function definition() {
+    public function definition(): void {
         global $DB;
 
         $webservicenames = $this->get_webservice_name_array();
 
         $mform = $this->_form;
 
         $options = [
-            'minchars' => 2,
-            'noselectionstring' => 'No webservices selected',
-            'multiple' => true,
-            'placeholder' => 'Search webservices...',
+            'minchars'          => 2,
+            'noselectionstring' => get_string('nowebservicesselected', 'tool_wsformat'),
+            'multiple'          => true,
+            'placeholder'       => get_string('searchwebservices', 'tool_wsformat'),
         ];
 
         // Documentation: https://docs.moodle.org/dev/lib/formslib.php_Form_Definition#autocomplete.
-        $mform->addElement('autocomplete', 'webservice_form', 'Webservices:', $webservicenames, $options);
+        $mform->addElement('autocomplete', 'webservice_form', get_string('webservices', 'tool_wsformat'), $webservicenames,
+         $options);
+
+        $buttonarray   = [];
+        $buttonarray[] = $mform->createElement('submit', 'submit', get_string('updateselection', 'tool_wsformat'));
+
+        $clearbutton   = '<button type="button" class="btn btn-secondary" '
+        .'onclick="window.location.href=\'index.php\'">'
```

Comment:
this also feels like it should just be a plain href attribute rather than inline JS

Reply:
and should have a full URL rather than a referential redirect